src/api.ts

```diff
@@ -2,7 +2,9 @@ import * as vscode from 'vscode';
 import { performAction } from './actions/client';
 import { ActionsController } from './actions/controller';
 import { ExecuteAction, SnippetAction, CommandAction, FileAction } from './actions/impl';
+import { getLogger } from './logger/logger';
 export * from "./actions/interfaces";
+import _ = require('lodash');
 
 export const bas = {
```

Comment:
1. is the type of `bas` defined anywhere?
2. evaluate separating the bas structure and the implementation of each method.
    basically what we have now does not scale, if we will have 20 methods/functions this api.ts file will be 1,000+ line long.

Reply:
to move my api to separate file? (thought about it) to create a folder for all apis?

Reply:
folder for all APIs seems like a good idea.
But start with only your new API (**don't** make this PR into a re-factor)